x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/json_diff/json_diff.test.tsx

```diff
@@ -0,0 +1,331 @@
+
+describe('Rule upgrade workflow: viewing rule changes in JSON diff view', () => {
+  it.each(['light', 'dark'] as const)(
+    'User can see precisely how property values would change after upgrade - %s theme',
+    (colorMode) => {
+      const oldRule: RuleResponse = {
+        ...savedRuleMock,
+      };
+
+      const newRule: RuleResponse = {
+        ...savedRuleMock,
+      };
+
+      /* Changes to test line update */
+      oldRule.version = 1;
+      newRule.version = 2;
+
+      /* Changes to test line removal */
+      oldRule.author = ['Alice', 'Bob', 'Charlie'];
+      newRule.author = ['Alice', 'Charlie'];
+
+      /* Changes to test line addition */
+      delete oldRule.license;
+      newRule.license = 'GPLv3';
+
+      const ThemeWrapper: React.FC<{}> = ({ children }) => (
+        <EuiThemeProvider colorMode={colorMode}>{children}</EuiThemeProvider>
+      );
+
+      const { container } = render(<RuleDiffTab oldRule={oldRule} newRule={newRule} />, {
+        wrapper: ThemeWrapper,
+      });
+
+      /*
+      Helper function to find an element with a particular text content.
+      React Testing Library's doesn't provide an easy way to search by text if text is split into multiple DOM elements.
+    */
+      const getChildByTextContent = (parent: Element, textContent: string): HTMLElement => {
+        return Array.from(parent.querySelectorAll('*')).find(
+          (childElement) => childElement.textContent === textContent
+        ) as HTMLElement;
+      };
+
+      /* LINE UPDATE */
+      const updatedLine = getChildByTextContent(container, '-  "version": 1+  "version": 2');
+
+      const updatedLineBefore = getChildByTextContent(updatedLine, '  "version": 1');
+      expect(updatedLineBefore).toHaveStyle(
+        `background: ${COLORS[colorMode].lineBackground.deletion}`
+      );
+
+      const updatedWordBefore = getChildByTextContent(updatedLineBefore, '1');
+      expect(updatedWordBefore).toHaveStyle(
+        `background: ${COLORS[colorMode].characterBackground.deletion}`
+      );
+
+      const updatedLineAfter = getChildByTextContent(updatedLine, '  "version": 2');
+      expect(updatedLineAfter).toHaveStyle(
+        `background: ${COLORS[colorMode].lineBackground.insertion}`
+      );
+
+      const updatedWordAfter = getChildByTextContent(updatedLineAfter, '2');
+      expect(updatedWordAfter).toHaveStyle(
+        `background: ${COLORS[colorMode].characterBackground.insertion}`
+      );
+
+      /* LINE REMOVAL */
+      const removedLine = getChildByTextContent(container, '-    "Bob",');
+
+      const removedLineBefore = getChildByTextContent(removedLine, '    "Bob",');
+      expect(removedLineBefore).toHaveStyle(
+        `background: ${COLORS[colorMode].lineBackground.deletion}`
+      );
+
+      const removedLineAfter = getChildByTextContent(removedLine, '');
+      expect(
+        removedLineAfter ? window.getComputedStyle(removedLineAfter).backgroundColor : null
```

Comment:
Not sure I understand why these ternary operators are needed here. How can `null` ever be `''`?

Reply:
Good catch! Looks like I forgot to change this line after refactoring.